omod/src/main/resources/webModuleApplicationContext.xml

```diff
@@ -27,6 +27,48 @@
 	<bean class="org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping"/>
 
 	<context:component-scan base-package="org.openmrs.module.fhirExtension" />
+	<bean id="taskServiceTarget" class="org.openmrs.module.fhirExtension.service.impl.TaskServiceImpl">
+		<property name="visitService" ref="visitService"/>
+		<property name="fhirTaskDao">

```

Comment:
any particular reason we arent using the [FhirTaskDao](https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirTaskDaoImpl.java)? Also, OMRS must have already used the above class, we can just inject that, instead of declaring a new instance here

Reply:
The FhirTaskDaoImpl is defined with @Component in the OMRS OMOD, so creating a new instance here for the same class `org.openmrs.module.fhir2.api.dao.impl.FhirTaskDaoImpl`
Can we refer the class through some other means ?

Reply:
Fixed this by using `fhirTaskDaoImpl` as reference name